src/http/client.rs

```diff
@@ -2168,11 +2334,17 @@ impl Http {
         T: Into<AttachmentType<'a>>,
     {
         let uri = api!("/channels/{}/messages", channel_id);
-        let url = match Url::parse(&uri) {
+        let mut url = match Url::parse(&uri) {
             Ok(url) => url,
             Err(_) => return Err(Error::Url(uri)),
         };
 
+        if let Some(proxy) = &self.proxy {
+            url.set_host(proxy.host_str()).map_err(|e| HttpError::Url(e))?;
+            url.set_scheme(proxy.scheme()).map_err(|_| HttpError::InvalidScheme)?;
+            url.set_port(proxy.port()).map_err(|_| HttpError::InvalidPort)?;
+        }
```

Comment:
Open to suggestions on refactoring this bit and the `proxy` field in general, feels a bit messy using `Url`.  Alternative way would be the previous version, just using `String` for the proxy field type and using a `String::replace` to create a new `Url`.

Reply:
Usage of `Url` is fine. It ensures the proxy is given with a valid url. Besides, this seems to be the only place where the url has to be changed to the proxy if one exists.